src/routes/Wallets/routes/Create/sagas/walletsCreate.js

```diff
@@ -0,0 +1,185 @@
+
+  if (!isPasswordExists) {
+    if (password !== passwordConfirm) {
+      yield put(setInvalidField('passwordConfirm', 'Password does not match confirmation'))
+
+      return
+    }
+
+    if (!passwordHint) {
+      yield put(setInvalidField('passwordHint', 'Password hint is required'))
+
+      return
+    }
+
+    if (password === passwordHint) {
+      yield put(setInvalidField('passwordHint', 'Password and hint should not be equal'))
+
+      return
+    }
+  }
+
+  yield walletsWorker.createRequest(walletsData, name, password, passwordHint)
+}
+
+function* createError(action: { payload: Error }): Saga<void> {
+  yield put(setInvalidField('password', action.payload.message))
+}
+
+function* createSuccess(action: WalletsSetWalletsAction): Saga<void> {
+  const { payload } = action
+
+  yield put(setWallets(payload))
+
+  const wallets: Wallets = payload.items
+  const createdWallet: Wallet = wallets[wallets.length - 1]
+
+  yield put(setActiveWallet(createdWallet.id))
+}
+
+export function* openView(): Saga<void> {
+  yield put(clean())
+}
+
+export function* closeView(): Saga<void> {
+  yield delay(config.delayBeforeFormClean)
+  yield put(clean())
+}
+
+export function* setNextStep(): Saga<void> {
+  const { currentStep }: WalletsCreateState = yield select(selectWalletsCreate)
+
+  switch (currentStep) {
+    case STEPS.NAME: {
+      yield checkName()
+      break
+    }
+
+    case STEPS.PASSWORD: {
+      yield create()
+      break
+    }
+
+    default:
+      break
+  }
+}
+
+function* goToWalletsStartView(): Saga<void> {
+  yield put(goToStartView())
+}
+
+function* goToWalletsCreateNameStep(): Saga<void> {
+  yield put(setCurrentStep(STEPS.NAME))
+}
+
+export function* setPrevStep(): Saga<void> {
+  const { currentStep }: WalletsCreateState = yield select(selectWalletsCreate)
+
+  switch (currentStep) {
+    case STEPS.NAME: {
+      yield goToWalletsStartView()
+      break
+    }
+
+    case STEPS.PASSWORD: {
+      yield goToWalletsCreateNameStep()
+      break
+    }
+
+    default:
+      break
+  }
+}
+
+export function* watchWalletsCreateOpenView(): Saga<void> {
+  yield takeEvery(OPEN_VIEW, openView)
+}
+
+export function* watchWalletsCreateCloseView(): Saga<void> {
+  yield takeEvery(CLOSE_VIEW, closeView)
+}
+
+export function* watchCreateWalletSetNextStep(): Saga<void> {
+  yield takeEvery(GO_TO_NEXT_STEP, setNextStep)
+}
+
+export function* watchCreateWalletSetPrevStep(): Saga<void> {
+  yield takeEvery(GO_TO_PREV_STEP, setPrevStep)
+}
+
+export function* watchCreateWalletCheckNameSuccess(): Saga<void> {
+  yield takeEvery(CHECK_NAME_SUCCESS, checkNameSuccess)
+}
+
+export function* watchCreateWalletCheckNameError(): Saga<void> {
+  yield takeEvery(CHECK_NAME_ERROR, checkNameError)
+}
+
+export function* watchCreateWalletCreateSuccess(): Saga<void> {
+  yield takeEvery(CREATE_SUCCESS, createSuccess)
+}
+
+export function* watchCreateWalletCreateError(): Saga<void> {
+  yield takeEvery(CREATE_ERROR, createError)
+}
```

Comment:
We should use single function to start sagas, like 
```
export function сrrateWalletRootSaga() {
  yield takeEvery(OPEN_VIEW, openView)
 ....
 yield takeEvery(CREATE_SUCCESS, createSuccess)
 yield takeEvery(CREATE_ERROR, createError)
}

```

Reply:
optional, would be more readable something like
```
....
 yield takeEvery(CREATE_ERROR, watchCreateWalletCreateError)
 yield takeEvery(CREATE_SUCCESS, watchCreateWalletCreateSuccess)
....
```

Reply:
https://github.com/jibrelnetwork/jwallet-web/blob/feature/update-import-wallet/src/routes/Wallets/routes/Create/sagas/walletsCreate.js#L134